lib/util.js

```diff
@@ -501,22 +508,47 @@ module.exports = {
           // TODO: have to add auth here (but first, auth to be put into the openapi tree)
         });
 
+        // If a folder has only one child which is a folder then we collapsed the child folder
+        // with parent folder.
+        /* eslint-disable max-depth */
+        if (resource.childCount === 1) {
+          for (let subChild in resource.children) {
+            let resourceSubChild = resource.children[subChild];
+            if (resourceSubChild.requests.length > 0) {
+              itemGroup.name += '/' + resourceSubChild.name;
```

Comment:
This would fail for the following case:
1. pets/a has x requests.
2. pets/a/b has y requests.
We would only get pets/a as the name of the collection/folder with x requests in it. All the y requests would be deleted